ipc_types/types.go

```diff
@@ -4,185 +4,192 @@ package ipc_types
 
 import (
 	"github.com/filecoin-project/go-address"
-	"github.com/ipfs/go-cid"
-	"math/big"
+	"golang.org/x/xerrors"
+	"path"
+	"strings"
 )
 
-// Go types for ipc-gateway/src/subnetid.rs, ipc-gateway/src/address.rs and ipc-subnet-actor and ipc-gateways' states and params
-
-// TODO ----------------------------------------------------------------------
-// TODO here below subnetid.rs and address.rs and related
-// TODO if moving this code to IPC gateway this could be taken from the same constant as written in Rust
-
 type IPCAddress struct {
 	SubnetID   SubnetID
 	RawAddress address.Address
 }
 
-type SubnetID struct {
-	Parent string
-	Actor  address.Address
+func newIPCAddressFromID(id uint64) *IPCAddress {
```

Comment:
We should remove this function, it leads to an invalid subnet address. This is not your fault, it is also the case in the Rust implementation, we will fix it there.

Reply:
ok